src/App.test.js

```diff
@@ -1,8 +1,32 @@
-import { render, screen } from '@testing-library/react';
+import { screen, render } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import userEvent from '@testing-library/user-event';
 import App from './App';
 
-test('renders learn react link', () => {
+test('renders page title', async () => {
   render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+  const pageTitle = await screen.findByRole('heading', { name: /Creatures of Hyrule/i });
+  expect(pageTitle).toBeInTheDocument();
+});
+
+test('renders creature cards', async () => {
+  render(<App />);
+  const headings = await screen.findAllByRole('heading');
+  expect(headings).toHaveLength(84);
+});
+
+test('should navigate to detail page', async () => {
+  render(
+    <MemoryRouter>
```

Comment:
Curious if you actually need MemoryRouter here since App already has access the routes.